### PR TITLE
perf(generic-metrics): Apply cardinality limits in separate thread

### PR DESCRIPTION
**Pull request by john-z-yang**
### Overview

Apply cardinality limits in a separate thread during cardinality rate limiting

### CODE REVIEW

**Review by nikhars — APPROVED** [review]
LGTM. Maybe wait for @untitaker approval before you merge.

**Review by untitaker — APPROVED** [review]

**Pull request merged**
